File: filemanager/src/main/java/com/merseyside/filemanager/mime_types/FileTypes.java
package com.merseyside.filemanager.mime_types;


import android.mtp.MtpConstants;

public enum FileTypes {

    PDF(new MimeType(getFormat(), "pdf", -1)),
    DOC(new MimeType(getFormat(), "msword", MtpConstants.FORMAT_MS_WORD_DOCUMENT)),
    XLS(new MimeType(getFormat(), "vnd.ms-excel", MtpConstants.FORMAT_MS_EXCEL_SPREADSHEET)),
    ZIP(new MimeType(getFormat(), "zip", -1));

    MimeType type;

    FileTypes(MimeType type) {
        this.type = type;
    }

    public MimeType getValue() {
        return type;
    }

    public static String getFormat() {
        return "application";
    }
}

File: filemanager/src/main/java/com/merseyside/filemanager/mime_types/FileTypes.kt
package com.merseyside.filemanager.mime_types

import android.mtp.MtpConstants

enum class FileTypes(var value: MimeType) {
    PDF(
        MimeType(format, "pdf", -1)
    ),
    DOC(
        MimeType(
            format,
            "msword",
            MtpConstants.FORMAT_MS_WORD_DOCUMENT
        )
    ),
    XLS(
        MimeType(format, "vnd.ms-excel", MtpConstants.FORMAT_MS_EXCEL_SPREADSHEET)
    ),
    ZIP(
        MimeType(format, "zip", -1)
    );

    companion object {
        val format: String
            get() = "application"
    }
}